fastlane/Fastfile

```diff
@@ -111,4 +117,28 @@ platform :ios do
       sh('bundle', 'exec', 'pod', 'repo', 'update')
     end
   end
+
+  desc "Retry pod push if sever error occured"
+  private_lane :retry_pod_push do |options|
+    pod_path = options[:pod_path]
+
+    server_error = " [!] An internal server error occurred"
+
+    retries = 5
+    i = 0
+
+    while (i < retries)
+      begin
+        sh('bundle', 'exec', 'pod', 'trunk', 'push', pod_path, '--allow-warnings', '--synchronous')
+        # if succeeded without error -> break out of loop
+        break
+      rescue => ex
+        unless ex.to_s.include? server_error
+          UI.error "Unrecoverable error in pod push trunk"
+          raise Exception.new ex
+        end
+        i = i + 1
+      end
+    end
+  end
```

Comment:
I'd suggest generalizing this.

```ruby
def with_retry(retries=5, wait=0.2)
  begin
    yield
  rescue Exception => e
    retries -= 1
    raise e if retries.zero?
    UI.error("Error occurred: #{exception}; retrying...")
    sleep(wait)
    retry
  end
```

We can use this as:

```ruby
with_retry do
  sh('bundle', 'exec', 'pod', 'trunk', 'push', pod_path, '--allow-warnings', '--synchronous')
end
```

I'd be tempted to drop the matching of error messages. These are liable to change. Might as well just retry in the face of errors.

Reply:
If we need to maintain the error message check, I'd suggest:

```ruby
with_retry do
  begin:
    sh('bundle', 'exec', 'pod', 'trunk', 'push', pod_path, '--allow-warnings', '--synchronous')
  rescue Exception => exception
    raise unless exception.message =~ /duplicate entry/i
    UI.warn("Version already exists (or something idk")
  end
end
```